src/MetaMorpho.sol

```diff
@@ -635,6 +631,20 @@ contract MetaMorpho is ERC4626, ERC20Permit, Ownable2Step, Multicall, IMetaMorph
         assets -= _simulateWithdrawMorpho(assets);
     }
 
+    /// @dev Returns the maximum amount of assets that the vault can supply on Morpho.
+    function _maxDeposit() internal view returns (uint256 totalSuppliable) {
```

Comment:
Argh nice catch

Reply:
How would it revert? The cap, even infinite, is limited to `type(uint184).max` while the supply queue is limited to `MAX_QUEUE_LENGTH` in size

Reply:
Oh right it's verified to be in the range of `uint184`. So that part is fine. There is still the part where those functions should return `type(uint256).max` when there is no deposit limit. In effect `type(uint184).max` is used to signify no limit, so should we comply to [that constraint from ERC4626](https://eips.ethereum.org/EIPS/eip-4626#:~:text=MUST%20return%202%20**%20256%20%2D%201%20if%20there%20is%20no%20limit%20on%20the%20maximum%20amount%20of%20assets%20that%20may%20be%20deposited.)

Reply:
Argh I hate this EIP because one could argue that the current code addresses the [constraint just above](https://eips.ethereum.org/EIPS/eip-4626#:~:text=MUST%20factor%20in%20both%20global%20and%20user%2Dspecific%20limits%2C%20like%20if%20deposits%20are%20entirely%20disabled%20(even%20temporarily)%20it%20MUST%20return%200.)... How do you feel about it?
I feel like it's fine because:
1. `type(uint184).max` can be considered like a global limit to the vault
2. it is complex to return `type(uint256).max` if all markets of the supply queue have an infinite cap (and if the curator inputs `type(uint184).max-1` as a market cap then this edge case check is flawed)

Reply:
else we just use `uint256` for the `cap` and too bad for the packing?

Reply:
Then it would overflow when summing up

Reply:
Ok so there is no "infinite cap" in metamorpho. The highest cap is `type(uint184).max` for each market, and it's actually not meaning that it can supply infinite to this market (even though in practice it's close to it). So I agree that it's fine to acknowledge